src/util.test.ts

```diff
@@ -232,6 +232,12 @@ describe('areNodesEqual', () => {
         test('document fragments', () => {
             expect(areNodesEqual(fragment, anotherFragment)).toBe(true)
         })
+        test('getAttributeNames fallback', () => {
```

Comment:
This test affects everything that comes after and tests only one situation. Could we instead run all the original `areNodesEqual` tests with and without the fallback, and restore the original implementation afterwards. It could be done as follows:

```
describe.each<[string, (() => string[]) | undefined]>([
    ['native', Element.prototype.getAttributeNames],
    ['undefined', undefined],
])(
    'areNodesEqual (getAttributeNames: %s)',
    (_: string, customGetAttributeNames: any) => {
        const originalGetAttributeNames = Element.prototype.getAttributeNames
        beforeAll(() => {
            Element.prototype.getAttributeNames = customGetAttributeNames
        })
        afterAll(() => {
            Element.prototype.getAttributeNames = originalGetAttributeNames
        })
        ...
```

Reply:
Fixed. Thanks for the details. Good learning.